SalesforceCommerceCloud/pwa-kit
# [V2] Port Environment Specific Configuration Support

# Description

This PR ports the recent work merged via #421, #445, and #438.

There is one important difference in this PR, and that is this implementation doesn't use a setter for setting the configuration value with in the SDK. Instead we are using conditional exports to provide a server implementation and a client implementation via a single file.

A deployed version of the PWA can be found [here](https://scaffold-pwa-development.mobify-storefront.com/)

# Types of Changes




# Changes

- Add configuration support to the runtime, surface the configuration via the `getConfig` function.
- Merge API configs in to new centralized config.
- Remove old pwa-kit.config.json files and references.

# How to Test-Drive This PR

- Test that custom configuration files work.
```
1. Create a new `local.js` config file in the config folder, this can be a copy of the default, but maybe change the URL settings so you can see a difference.
2. Preview the App `npm run start`
3. You should see the reflected changes as your application uses the `local.js` config over the default.
```
- Test that generating projects works still
```
1. Generated a project using this command `GENERATOR_PRESET=test-project node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/project`
2. Preview the App `npm run start`
3. Smoke test that the app is working.
4. Try making config changes like in part 1 with the generated project. You should have the same result.
```

# Checklists




## General
